### PR TITLE
feat: implement non-interactive default rules for reduced padding

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/1149

### CODE REVIEW

**Comment by rootulp** [comment]
💀 https://github.com/celestiaorg/celestia-app/actions/runs/3761251730/attempts/1

**Comment by evan-forbes** [comment]
yeah, we should run the fuzzers locally for a while on main and then on #1147 and see if we get failures on each. naively it makes sense that we would get failures without #1147 after #1137

**Comment by rootulp** [comment]
> naively it makes sense that we would get failures without https://github.com/celestiaorg/celestia-app/pull/1147 after https://github.com/celestiaorg/celestia-app/pull/1137

+1, hoping it gets resolved via #1137

**Comment by evan-forbes** [comment]
I ran tests for a over an hour with no failures locally in #1154 after merging #1147, so I think we're back to no flakes :crossed_fingers:

File: pkg/shares/non_interactive_defaults.go
Comment (evan-forbes) [inline pkg/shares/non_interactive_defaults.go:89]: [non-blocking question]
where's the other one?

**Review by evan-forbes — COMMENTED** [review]
will take a second look tmrw, but LGTM so far

File: pkg/shares/non_interactive_defaults.go
Comment (rootulp) [reply pkg/shares/non_interactive_defaults.go:89]: https://github.com/celestiaorg/celestia-app/blob/b65353f456335ed195c67522d4ec546dc3a583fb/x/blob/types/payforblob.go#L178-L182

**Review by evan-forbes — APPROVED** [review]
this LGTM, but might want to have @nashqueue take a look just to have a second pair of eyes

I'm also okay with merging this now, and @nashqueue can inspect after, as it would be nice to know if it breaks #1154

**Comment by rootulp** [comment]
sending it 🚀 can address any other PR feedback in a follow-up PR.

File: app/write_square_test.go
Comment (nashqueue) [inline app/write_square_test.go:93]: [non-blocking][nit]
```suggestion
			expectedIndexes: []uint32{
				// BlobMinSquareSize(2) = 2 so the first blob has to start at the
				// next multiple of 2 > 32 which is 32. This blob occupies
				// shares 32 to 33.
				32,
				// BlobMinSquareSize(20) = 8 so the second blob has to start at
				// the next multiple of 8 > 34 which is 40. This blob occupies
				// shares 40 to 59
				40,
				// BlobMinSquareSize(199) = 16 so the third blob has to start at
				// the next multiple of 16 > 48 which is 64. This blob occupies
				// shares 64 to 262.
				64,
			},
```
If the blob occupies shares 2 to 4 it would occupy 3 shares in my understanding. Also depends on how you start counting 0 or 1. I started counting at 0 for this example. The last share of the square would be 263 in this case. The last square of the blob would be 262.

**Review by nashqueue — COMMENTED** [review]
LGTM

File: app/write_square_test.go
Comment (rootulp) [reply app/write_square_test.go:93]: Agreed, thanks for pointing this out! This example assumes 0-based indexing. The last index in the ranges are all off by one so I'll update to your suggestion in a follow-up PR.

Comment (nashqueue) [reply app/write_square_test.go:93]: heads up the second one is off by 11

Comment (rootulp) [reply app/write_square_test.go:93]: Agreed, it wasn't just off by one. Fixed in https://github.com/celestiaorg/celestia-app/pull/1185

**Pull request merged**
